app/actions.ts

```diff
@@ -107,6 +112,34 @@ export async function clearChats() {
   return redirect('/')
 }
 
+export async function deleteSavedFiles(chatId: string, userId: string, path?: string) {
+  const session = await auth()
+
+  if (session?.user?.id !== userId) {
+    return {
+      error: 'Unauthorized'
+    }
+  }
+  const chat = await getChat(chatId, userId) as Chat
+  if (!chat || 'error' in chat) {
+    return redirect('/')
+  }
+
+  const deleteFileUrls = []
+  for (const message of chat.messages) {
+    if (message.files) {
+      const files = JSON.parse(message.files || '')
```

Comment:
Why 
```
const files = JSON.parse(message.files || '')
```
Instead of this shouldn't we be checking for `message.files` for null

Reply:
yes you're right i believe that was part of older error checking, updated